### DISCUSSION TITLE
why rook only support set "user quota", not support "bucket quota"?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#10423] [Q&A]
![image](https://user-images.githubusercontent.com/13997135/172802356-a78666c6-013c-4ad3-b283-7c1578ea1d7c.png)

**Comment by zhucan** [comment]
go-ceph only support set "user quota".

**Reply by travisn** [reply]
You're referring to the [object user quotas](https://rook.github.io/docs/rook/v1.9/CRDs/Object-Storage/ceph-object-store-user-crd/) that exists? 
@BlaineEXE Are there quotas on OBCs?

**Comment by zhucan** [comment]
The quotas set on the obcs was used to user quota.

**Reply by thotz** [reply]
Yes, OBC does not equivalent to a backend bucket, it is more like an access to the backend rgw bucket. The same rgw bucket can be shared with different OBCs. So what I meant was mapping between OBC and backend RGW bucket is not 1:1. Hence we are setting a quota on the internal ceph user created for the OBC

**Comment by zhucan** [comment]
But we still need to set quota to bucket, because of the s3 policy, one bucket can be used by not only one user, the user and bucket is not 1:1.

**Reply by thotz** [reply]
Currently, sorry we don't have support in CR to set bucket quota, need to do that manually via radosgw-admin cli/ adminops https://docs.ceph.com/en/latest/radosgw/admin/#set-bucket-quota

**Reply by zhucan** [reply]
go-ceph only support set quota to user

**Reply by thotz** [reply]
Yaa thats true go-ceph have support for user-quota, but we can add those apis if needed. But still radosgw-admin cli works

**Reply by zhucan** [reply]
@thotz Thanks
